ark/utils/load_utils.py

```diff
@@ -319,3 +321,197 @@ def load_imgs_from_dir(data_dir, files=None, match_substring=None, trim_suffix=N
+        r = re.compile('.*R.*C.*')
+        if r.match(dir) is None:
+            raise ValueError(f"Invalid FOVs found in directory, {data_dir}. FOV folder names "
+                             f"should have the form RnCm.")
+    if len(fov_list) == 0:
+        raise ValueError(f"No FOVs found in directory, {data_dir}.")
+
+    expected_fovs = get_tiled_fov_names(fov_list)
+
+    # get image size if not provided
+    if not max_image_size:
+        max_image_size = get_max_img_size(data_dir, img_sub_folder, fov_list)
+
+    # get imgs from first fov if no img names supplied
+    if channels is None:
+        channels = iou.list_files(
+            dir_name=os.path.join(data_dir, fov_list[0], img_sub_folder),
+            substrs=['.tif', '.jpg', '.png']
+        )
+        channels.sort()
+    # otherwise, fill channel names with correct file extension
+    elif not all([img.endswith(("tif", "tiff", "jpg", "png")) for img in channels]):
+        # need this to reorder channels back because list_files may mess up the ordering
+        channels_no_delim = [img.split('.')[0] for img in channels]
+
+        all_channels = iou.list_files(
+            dir_name=os.path.join(data_dir, fov_list[0], img_sub_folder),
+            substrs=channels_no_delim, exact_match=True
+        )
+
+        # get the corresponding indices found in channels_no_delim
+        channels_indices = [channels_no_delim.index(chan.split('.')[0]) for chan in all_channels]
+
+        # verify if channels from user input are present in `all_channels`
+        all_channels_no_delim = [channel.split('.')[0] for channel in all_channels]
+
+        misc_utils.verify_same_elements(all_channels_in_folder=all_channels_no_delim,
+                                        all_channels_detected=channels_no_delim)
+        # reorder back to original
+        channels = [chan for _, chan in sorted(zip(channels_indices, all_channels))]
+
+    if len(channels) == 0:
+        raise ValueError("No images found.")
+
+    # no missing fov images, load data normally
+    if len(fov_list) == len(expected_fovs):
+        img_xr = load_imgs_from_tree(data_dir, img_sub_folder, fovs=fov_list, channels=channels,
+                                     max_image_size=max_image_size)
+        return img_xr
+    # missing fov directories
+    else:
+        test_img = io.imread(os.path.join(data_dir, fov_list[0], img_sub_folder, channels[0]))
+        dtype = test_img.dtype
+
+    img_data = np.zeros((len(expected_fovs), max_image_size, max_image_size, len(channels)),
+                        dtype=dtype)
+
+    for fov, fov_name in enumerate(expected_fovs):
+        # load in fov data for images, leave missing fovs as zeros
+        if fov_name in fov_list:
+            fov_digits = re.findall(r'\d+', fov_name)
+            start_row = (int(fov_digits[0]) - 1) * max_image_size
+            start_col = (int(fov_digits[1]) - 1) * max_image_size
+
+            for img in range(len(channels)):
```

Comment:
Same thing here, if this function only ever takes in one channel this code gives a misleading idea of what this function does